run_diff.go

```diff
@@ -147,10 +153,26 @@ func diffResults(before map[string][]int, after map[string][]int) map[string][]i
 		}
 	}
 	for test, resultsAfter := range after {
+		if paths != nil && !anyPathMatches(paths, test) {
```

Comment:
I think the reason for the nil check here and not above is intentional, but that would be clearer if both were handled outside of anyPathMatches, or with a comment.

Reply:
Not deliberate. Fixed.